yt_dlp/extractor/nubilesporn.py

```diff
@@ -0,0 +1,143 @@
+            'webpage_url': f'{_MEMBERS_URL}/video/watch/165320/trying-to-focus-my-one-track-mind-s3e1',
+            'categories': ['Big Boobs', 'Big Naturals', 'Blowjob', 'Brunette', 'Cowgirl', 'Girl Orgasm', 'Girl-Boy',
+                           'Glasses', 'Hardcore', 'Milf', 'Shaved Pussy', 'Tattoos', 'YoungerMommy.com'],
+            'tags': list,
+            'cast': ['Kenzie Love'],
+            'availability': 'needs_auth',
+            'series': 'Younger Mommy',
+            'series_id': 64,
+            'season': 'Season 3',
+            'season_number': 3,
+            'episode': 'Episode 1',
+            'episode_number': 1
+        }
+    }]
+
+    def _perform_login(self, username, password):
+        login_webpage: str = self._download_webpage(f'{NubilesPornIE._BASE_URL}/login', video_id=None)
+        hidden_inputs: dict = self._hidden_inputs(login_webpage)
+        hidden_inputs.update({'username': username, 'password': password})
+        self._request_webpage(f'{NubilesPornIE._BASE_URL}/authentication/login',
+                              data=urlencode_postdata(hidden_inputs), video_id=None)
+
+    @staticmethod
+    def _get_channel_info(element: str) -> dict:
+        info = {}
+        path = extract_attributes(get_element_html_by_class('site-link', element)).get('href')
+        if path:
+            info['url'] = f'{NubilesPornIE._MEMBERS_URL}{path}'
+            info['id'] = int_or_none(re.findall('/([0-9]+)$', path)[0])
+        raw_name = get_element_by_class('site-link', element)
+        if raw_name:
+            info['name'] = ' '.join(re.findall('[A-Z][^A-Z]*', raw_name.replace('.com', '')))
+        return info
+
+    @staticmethod
+    def _get_formats(element) -> list:
+        return [NubilesPornIE._get_format(i) for i in get_elements_html_by_class('edge-download-item', element)]
+
+    @staticmethod
+    def _get_format(element: str) -> dict:
+        res = parse_resolution(get_element_text_and_html_by_tag('div', element)[1])
+        url = extract_attributes(get_element_html_by_class('btn', element)).get('href')
+        return dict(url=f'https:{url}', **res, format_id=f'mp4_{res.get("width")}_{res.get("height")}')
+
+    @staticmethod
+    def _get_categories(element: str) -> Optional[list]:
+        raw_category_section = get_element_by_class('categories', element) or ""
+        raw_categories = get_elements_by_class('btn', raw_category_section)
+        return [clean_html(i) for i in raw_categories]
+
+    @staticmethod
+    def _get_tags(element: str) -> Optional[list]:
+        raw_tag_section = get_elements_by_class('tags', element)
+        if len(raw_tag_section) >= 2:
+            raw_tags = get_elements_by_class('btn', raw_tag_section[1])
+            return [clean_html(i) for i in raw_tags]
+
+    def _real_extract(self, url):
+        url_match = self._match_valid_url(url)
+        video_id = url_match.group('id')
+        page: str = self._download_webpage(url, video_id=video_id)
```

Comment:
```suggestion
        video_id = self._match_valid_url(url).group('id')
        page = self._download_webpage(url, video_id)
```

Reply:
I've kept the `url_match` declaration as-is, as it is used a few times. The same applies to `media_entries`, `channel_name`, and `channel_id`.